src/WikiService.Validation/Rubric/EditRubricRequestValidator.cs

```diff
@@ -0,0 +1,195 @@
+
+              if (opValue is null)
+              {
+                return !await _rubricRepository.DoesRubricNameExistAsync(
+                  null,
+                  x.Item2.Operations.FirstOrDefault(
+                    o => o.path.EndsWith(nameof(EditRubricRequest.Name), StringComparison.OrdinalIgnoreCase)).value?.ToString());
+              }
+              else if (Guid.TryParse(opValue, out Guid parentId))
+              {
+                return !await _rubricRepository.DoesRubricNameExistAsync(
+                  parentId,
+                  x.Item2.Operations.FirstOrDefault(
+                    o => o.path.EndsWith(nameof(EditRubricRequest.Name), StringComparison.OrdinalIgnoreCase)).value?.ToString());
+              }
+              
+              return false;
+            })
+            .WithMessage("Name already exists.");
+        });
+
+      When(x => !x.Item2.Operations.Any(o => o.path.EndsWith(nameof(EditRubricRequest.Name), StringComparison.OrdinalIgnoreCase))
+        && x.Item2.Operations.Any(o => o.path.EndsWith(nameof(EditRubricRequest.ParentId), StringComparison.OrdinalIgnoreCase)),
+        () =>
+        {
+          RuleFor(x => x)
+            .MustAsync(async (x, _) =>
+            {
+              DbRubric oldRubric = await _rubricRepository.GetAsync(x.Item1);
+              string rubricName = oldRubric.Name;
+              string opValue = x.Item2.Operations.FirstOrDefault(
+                  o => o.path.EndsWith(nameof(EditRubricRequest.ParentId), StringComparison.OrdinalIgnoreCase)).value?.ToString();
+
+              if (opValue is null)
+              {
+                return !await _rubricRepository.DoesRubricNameExistAsync(null, rubricName);
+              }
+              else if (Guid.TryParse(opValue, out Guid parentId))
+              {
+                return !await _rubricRepository.DoesRubricNameExistAsync(parentId, rubricName);
+              }
+              
+              return false;
+            })
+            .WithMessage("Name already exists.");
+        });
+
+      When(x => !x.Item2.Operations.Any(o => o.path.EndsWith(nameof(EditRubricRequest.ParentId), StringComparison.OrdinalIgnoreCase))
+        && x.Item2.Operations.Any(o => o.path.EndsWith(nameof(EditRubricRequest.Name), StringComparison.OrdinalIgnoreCase)),
+        () =>
+        {
+          RuleFor(x => x)
+            .MustAsync(async (x, _) =>
+            {
+              DbRubric oldRubric = await _rubricRepository.GetAsync(x.Item1);
+              Guid? rubricParentId = oldRubric.ParentId;
+              
+              return !await _rubricRepository.DoesRubricNameExistAsync(
+                rubricParentId,
+                x.Item2.Operations.FirstOrDefault(
+                  o => o.path.EndsWith(nameof(EditRubricRequest.Name), StringComparison.OrdinalIgnoreCase)).value?.ToString());
+            })
+            .WithMessage("Name already exists.");
+        });
```

Comment:
try to gether three types of When
When(x => !x.Item2.Operations.Any
(o => o.path.EndsWith(nameof(EditRubricRequest.ParentId), StringComparison.OrdinalIgnoreCase) || o.path.EndsWith(nameof(EditRubricRequest.Name), StringComparison.OrdinalIgnoreCase))

and make a common validation for all this cases.
so, you need two checks (inside when):
does ParentId changes (if changes, use new Id; if don't - use old)
does Name changes (if changes, use new Name; if don't - use old)

and than use DoesRubricNameExistAsync ones with right params (new or old ParentId and new or old Name)